### PR TITLE
Support to add custom annotations and labels for CR

### LINKED ISSUE #930: [sn-platform] Support to add custom annotations and labels for CR
# Motivation
There are some feature flags triggered by specific annotations or labels.
So the sn-platform should support to add custom annotations and labels for CR, such as PulsarBroker, PulsarProxy. 

```yaml
apiVersion: pulsar.streamnative.io/v1alpha1
kind: PulsarBroker
metadata:
  name: "ok-sn-platform"
  namespace: default
  annotations:
    "cloud.streamnative.io/omit-init-sysctl": "true"
```

### PR DESCRIPTION
**Pull request by FushuWang**
Fixes #930 

### Modifications

Add values for PulsarBroker, PulsarProxy, BookkeeperCluster, and ZookeeperCluster.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform/values.yaml [hunk 1]
```diff
@@ -827,6 +834,13 @@ zookeeper:
 ## templates/bookkeeper-statefulset.yaml
 ##
 bookkeeper:
+  # bookkeeperCluster indicates the setup for the CR(CustomResource) of the BookkeeperCluster
+  bookkeeperCluster:
```
Comment (ericsyh) [inline charts/sn-platform/values.yaml:838]: Shall we use `BookKeeperCluster` to keep consistent with the CR?

Comment (FushuWang) [reply charts/sn-platform/values.yaml]: Fixed.

**Review by maxsxu — APPROVED** [review]

File: charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml [hunk 2]
```diff
@@ -25,12 +25,18 @@ metadata:
...
+    {{- with .Values.bookkeeper.bookKeeperCluster.annotations -}}
+    {{ toYaml . | nindent 4 }}
+    {{- end }}
   labels:
     {{- include "pulsar.standardLabels" . | nindent 4 }}
     component: {{ .Values.bookkeeper.component }}
 {{- with .Values.bookkeeper.labels }}
 {{ toYaml . | indent 4 }}
 {{- end }}
+    {{- with .Values.bookkeeper.bookKeeperCluster.labels -}}
```
Comment (ciiiii) [inline charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml:37]: How about using the same indent format of `.Values.bookkeeper.labels` to make the template neat?

Comment (FushuWang) [reply charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml]: Fixed

**Review by ericsyh — APPROVED** [review]

**Comment by ericsyh** [comment]
@ciiiii PTAL again

**Pull request merged**